src/components/AccessCodeModal.test.tsx

```diff
@@ -0,0 +1,102 @@
+/*
+ * SPDX-FileCopyrightText: 2024 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { act, screen } from '@testing-library/react';
+import { expect, it, vi, describe } from 'vitest';
+
+import { AccessCodeModal } from './AccessCodeModal';
+import { setup } from '../test/utils';
+
+describe('AccessCodeModal', () => {
+	it('should display modal title, a description and a confirmation button', async () => {
+		setup(<AccessCodeModal wrongAccessCode={false} queryWithAccessCode={vi.fn()} />);
+
+		await act(async () => {
+			await vi.advanceTimersByTimeAsync(1);
+		});
+
+		expect(screen.getByText('The link is secured by an access code')).toBeVisible();
+		expect(screen.getByText('Please, insert the access code to view the folder')).toBeVisible();
+		expect(screen.getByText('Done')).toBeVisible();
+	});
+
+	it('should display access code input with an icon', async () => {
+		setup(<AccessCodeModal wrongAccessCode={false} queryWithAccessCode={vi.fn()} />);
+
+		await act(async () => {
+			await vi.advanceTimersByTimeAsync(1);
+		});
+
+		expect(screen.getByLabelText(/access code/i)).toBeVisible();
+		expect(screen.getByTestId('icon: EyeOffOutline')).toBeVisible();
+	});
+
+	it('should change the input type from password to text when the user clicks on the eye icon of the input', async () => {
+		const { user } = setup(
+			<AccessCodeModal wrongAccessCode={false} queryWithAccessCode={vi.fn()} />
+		);
+
+		await act(async () => {
+			await vi.advanceTimersByTimeAsync(1);
```

Comment:
I would prefer having the TIMERS.showModal/modalDelay const instead of the number inside the advanceTimers, it would be immediately clear why the advanceTimers is needed